Add a diagnostic script

milljm opened the pull request:
Created a simple diagnostic script which can be executed
to print useful system information to stdout.

Modified update_and_rebuild_libmesh.sh script to make use
of diagnostics.sh.

Closes #11255

Commit:
Add a diagnostic script

Created a simple diagnostic script which can be executed
to print useful system information to stdout.

Modified update_and_rebuild_libmesh.sh script to make use
of diagnostics.sh.

Closes #11255

milljm commented on scripts/diagnostics.sh:
```diff
@@ -0,0 +1,111 @@
+    fi
+  fi
+}
+
```
file_born_date is not used in this script... but _may_ be useful in the future...

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/176436/) on 6e17ec9 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11260/site)

This comment will be updated on new commits.

Commit:
Adding copyright header

Forgot to add copyright header

Closes #11255

permcody reviewed (dismissed):
This looks OK to me. I'll leave it for Brian to take a second look at.

Commit:
move date print out

Move the routine of printing the date when the script is
launched to near where the other diagnostics are executing.

Closes #11255
